### PR TITLE
added test to indexing on groupby, #32464

### LINKED ISSUE #32464: Grouping by all columns of an empty DataFrame should produce MultiIndex, but doesn't
Greetings!

This comes up in [cuDF](https://github.com/rapidsai/cudf) in our tests that check for compatibility with Pandas:

```
In [21]: a = pd.DataFrame({'a': [], 'b': [], 'c': []})

In [22]: a.groupby(['a', 'b']).sum().index
Out[22]: MultiIndex([], names=['a', 'b'])

In [23]: a.groupby(['a', 'b', 'c']).sum().index
Out[23]: Index([], dtype='object')
```

Would it be more consistent for grouping by multiple columns to always result in a MultiIndex?

Edit: Checked with Pandas 1.0.1 as well as 0.25.3 -- same behaviour.

### PR DESCRIPTION
**Pull request by gabrieldi95**
- [x] closes #32464
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

This makes sure an index of a groupby with more then one column is always a MultiIndex.

### CODE REVIEW

**Comment by gabrieldi95** [comment]
I tried using `tm.assert_frame_equal` to test if a.groupby(["a", "b", "c"]).sum() has a MultiIndex like this:

```python
    a = DataFrame({"a": [], "b": [], "c": []})
    result = a.groupby(["a", "b", "c"]).sum()

    expected_index = MultiIndex.from_arrays([[], [], []], names=("a", "b", "c"))
    expected = DataFrame(
        np.ndarray((0, 0)), dtype="float64", index=expected_index, columns=Index([])
    )

    # Tests if groupby with all columns has a multiindex
    tm.assert_frame_equal(result, expected)
```

But it didn't work, so I'm keeping just the other assertion, that tests if both groupby's are equel.

**Comment by jreback** [comment]
can you merge master and update to comments

**Comment by gabrieldi95** [comment]
Yes, I was working on other stuff, will work on it now!

**Comment by gabrieldi95** [comment]
Hm I don't really know why the check fail, can someone help me?

**Comment by jreback** [comment]
can you merge master again and ping on green

**Comment by gabrieldi95** [comment]
@jreback can you help me with the error?

**Comment by jreback** [comment]
is worthwhile but needs to be passing (and pls merge master)

**Comment by rhshadrach** [comment]
@gabrieldi95 - The CI failure appears unrelated to me; can you try merging master again

**Comment by jreback** [comment]
@gabrieldi95 can you merge master and address comments

**Comment by rhshadrach** [comment]
Closing to take this off the queue. @gabrieldi95 please let me know if you'd like to continue this!

**Comment by gabrieldi95** [comment]
@rhshadrach sorry for taking so long! Yes, I would like to continue

File: pandas/tests/groupby/test_function.py [hunk 1]
```diff
@@ -1155,3 +1155,21 @@ def test_groupby_sum_below_mincount_nullable_integer():
...
     expected = DataFrame({"b": [pd.NA] * 3, "c": [pd.NA] * 3}, dtype="Int64", index=idx)
     tm.assert_frame_equal(result, expected)
+
+
+def test_if_is_multiindex():
+    # GH 32464
+    # Test if index after groupby with more then one column is always MultiIndex
+    a = DataFrame({"a": [], "b": [], "c": []})
+
+    agg_1 = a.groupby(["a", "b"]).sum()
```
[the remark this answers is not in the document]
Comment (rhshadrach) [reply pandas/tests/groupby/test_function.py:1175]: Can you just test in the case of an empty frame here - nonempty is covered elsewhere. Indeed it is tricky to construct an empty MultiIndex of the expected type; this works:

```
df = DataFrame({"a": [1], "b": [2], "c": [3]}).set_index(['a', 'b', 'c'])
result = df.groupby(["a", "b", "c"]).sum().index[:0]
expected = df.index[:0]
tm.assert_index_equal(result, expected)
```

File: pandas/tests/groupby/test_function.py [hunk 2]
```diff
@@ -1167,6 +1167,38 @@ def test_groupby_sum_below_mincount_nullable_integer():
     tm.assert_frame_equal(result, expected)
 
 
+def test_if_is_multiindex():
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_function.py:1170]: Can you rename to something like test_empty_multiindex

**Review by rhshadrach — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you merge master

**Comment by jreback** [comment]
can you merge master and update to comments

File: pandas/tests/groupby/test_function.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/tests/groupby/test_function.py:1179]: @gabrieldi95 can you replace the test with basically this example

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in main and address the review and we can reopen.

**Pull request closed without merging**